test/context_test.go

```diff
@@ -220,20 +220,28 @@ func testContextRequestWithCancel(t *testing.T, nc *nats.Conn) {
 		cancelCB()
 	})
 
-	nc.Subscribe("slow", func(m *nats.Msg) {
+	sub1, err := nc.Subscribe("slow", func(m *nats.Msg) {
 		// simulates latency into the client so that timeout is hit.
 		time.Sleep(40 * time.Millisecond)
 		nc.Publish(m.Reply, []byte("OK"))
 	})
-	nc.Subscribe("slower", func(m *nats.Msg) {
+	if err != nil {
+		t.Fatalf("Expected to be able to subscribe: %s", err)
+	}
+	defer sub1.Unsubscribe()
+	sub2, err := nc.Subscribe("slower", func(m *nats.Msg) {
 		// we know this request will take longer so extend the timeout
 		expirationTimer.Reset(100 * time.Millisecond)
 
 		// slower reply which would have hit original timeout
-		time.Sleep(90 * time.Millisecond)
+		time.Sleep(70 * time.Millisecond)
```

Comment:
What's the reason for this change here?

Reply:
This test depends on request either timing out or not depending on the circumstances. The expiration timer is reset in each subscribe callback to 100ms - sleeping for 90ms was making it fail quite often if there the subsequent request had 10ms delay. Now it's 30ms "buffer", which makes it less, but still, flaky...

I get why this test is useful but its tough to fix it entirely while retaining the logic.

Reply:
Oh, gotcha. That feels pretty flaky no matter what we set here, seeing faster or slower machines running it...